NOTICE

```diff
@@ -1,5 +1,5 @@
-Polaris
-Copyright 2024 Snowflake Computing Inc.
+Apache Polaris (incubating)
+Copyright 2024 The Apache Software Foundation
```

Comment:
This PR replaces copyright headers that credit Snowflake with the standard ASF header. When we do that, I think [ASF policy](https://www.apache.org/legal/src-headers.html#headers) requires that someone from the company that is donating submit or approve the header changes. Here's the specific guidance:

> 1. If the source file is submitted with a copyright notice included in it, the copyright owner (or owner's agent) must either:
>     a. remove such notices, or
>     b. move them to the NOTICE file associated with each applicable project release, or
>     c. provide written permission for the ASF to make such removal or relocation of the notices.

What I've done in the past is had someone from the donating company approve the PR and taken option b to relocate the copyright header to NOTICE ([example](https://github.com/apache/iceberg/blob/main/NOTICE#L10-L25)). That is usually an easy solution, but it does have a downside that it adds NOTICE content that has to be included in downstream NOTICE files. When we donated Iceberg, I was working for Netflix and removed the headers so we didn't need to include it in NOTICE.

Also, the NOTICE below about Apache Iceberg is no longer required for ASF, and was specifically discouraged by Roy in a comment. It's fine to keep it, but you may want to remove it.

Reply:
For the context, I added this file before the move to ASF Incubator.
Snowflake is preparing SGA + gonna review this PR to approve header change (I discussed with @takidau about that).
For Iceberg mention in NOTICE, you are right, I forgot to remove it (it was required before ASF Incubator, not needed anymore in the incubator).

Reply:
As we have SGA, and the code is fully donated, I don't think we need to mention the original NOTICE.

I updated the NOTICE, and let me assign @takidau and @annafil for review and merge.